src/theory/datatypes/theory_datatypes.cpp

```diff
@@ -631,8 +631,9 @@ Node TheoryDatatypes::expandDefinition(LogicRequest &logicRequest, Node n) {
       }
       else
       {
-        Assert(t.isRecord());
-        const Record& record = t.getRecord();
+        Assert(t.toType().isRecord());
+        const Record& record =
+            static_cast<DatatypeType>(t.toType()).getRecord();
```

Comment:
(see other comments about `static_cast<DatatypeType>`)

```suggestion
            DatatypeType(t.toType()).getRecord();
```